extensions/app-configuration/runtime/src/main/java/io/quarkiverse/azure/app/configuration/AzureAppConfigurationConfig.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkiverse.azure.app.configuration;
+
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+
+@ConfigMapping(prefix = "quarkus.azure.app.configuration")
+@ConfigRoot(phase = ConfigPhase.RUN_TIME)
+public interface AzureAppConfigurationConfig {
+    /** The endpoint of the app configuration */
+    String endpoint();
+
```

Comment:
Is this really supposed to work? All Config types I have seen in Quarkus so far were classes, not interfaces.

Reply:
Yes, this is now the recommended way to write configuration for extensions. The idea was to align APIs between Quarkus and App development: https://quarkus.io/guides/writing-extensions#configuration

Reply:
Thanks, I was not aware of that.